### PR TITLE
Bloody Scorpion Theme

### PR DESCRIPTION
**Pull request by JSanchoG**

### CODE REVIEW

**Comment by JSanchoG** [comment]
I added `image.png` file, you said you could add blur and compress it to `.webp`.

**Comment by nstringham** [comment]
@JSanchoG You know you can change your old PR instead of opening a new one right?

**Comment by JSanchoG** [comment]
> @JSanchoG You know you can change your old PR instead of opening a new one right? 

I ate 3 kids

**Comment by nstringham** [comment]
> > @JSanchoG You know you can change your old PR instead of opening a new one right?
> 
> I ate 3 kids

You what!

**Review by JSanchoG — COMMENTED** [review]
yes.

**Comment by nstringham** [comment]
## There are some technical and some content problems with this PR

### Technical problems
these problems are a technical error
- background type `linear-gradient` doesn't work
I would highly recommend you clone your branch and test things before making PRs with Code changes
see [CONTRIBUTING.md](https://github.com/SkyCryptWebsite/SkyCrypt/blob/master/CONTRIBUTING.md) for how to get SkyCyrpt running locally for development

### Content problems
these problems are not errors but they make the theme not fit in well with the other skycrypt themes
- `link` is a different color than all the other colors
- `hover` is darker than `link`

File: public/resources/ts/themes.ts [hunk 1]
```diff
@@ -68,6 +68,9 @@ export function loadTheme(currentTheme: string): void {
           })
           .join(", ")})`;
         break;
+      case "linear-gradient":
+        value = 'linear-gradient(${background.angle} 0%, ${background.colors} ${background.end})';
```
Comment (nstringham) [inline public/resources/ts/themes.ts:72]: fix `linear-gradient`

File: src/constants/themes.js [hunk 2]
```diff
@@ -155,6 +155,35 @@ module.exports = {
...
+          type: "linear-gradient",
+          angle: "90deg",
+          end: "100%",
+          colors: ["#fdbb3c", "#F09819"],
+        },
+      },
+      colors: {
+        logo: "#ED1444",
+        icon: "#ED1444",
+        link: "#38ef7d",
```
Comment (nstringham) [inline src/constants/themes.js:181]: make `link` red

**Review by nstringham — CHANGES_REQUESTED** [review]

**Comment by JSanchoG** [comment]
> ## There are some technical and some content problems with this PR
> ### Technical problems
> these problems are a technical error
> 
> * background type `linear-gradient` doesn't work
>   I would highly recommend you clone your branch and test things before making PRs with Code changes
>   see [CONTRIBUTING.md](https://github.com/SkyCryptWebsite/SkyCrypt/blob/master/CONTRIBUTING.md) for how to get SkyCyrpt running locally for development
> 
> ### Content problems
> these problems are not errors but they make the theme not fit in well with the other skycrypt themes
> 
> * `link` is a different color than all the other colors
> * `hover` is darker than `link`

colors fixed, but i don't know anything about TypeScript.

**Comment by nstringham** [comment]
> i don't know anything about TypeScript
It's basically just JavaScript

**Review by nstringham — COMMENTED** [review]
92d753506fb69af31e86391aa72bbc52163382a0 should be reverted because it makes things worse

**Comment by JSanchoG** [comment]
> [92d7535](https://github.com/SkyCryptWebsite/SkyCrypt/commit/92d753506fb69af31e86391aa72bbc52163382a0) should be reverted because it makes things worse

I removed the `linear-gradient`, because I cannot connect to MongoDB.
I think that's it now.

**Review by JSanchoG — COMMENTED** [review]
nevermind, i think i fixed the gradient

**Comment by JSanchoG** [comment]
@nstringham i also did this, but idk how to implement it (lvl 100 pets background blur box-shadow)
Przechwytywanie
```
<!-- Level 100 Pets -->
<!-- Level 100 Pets -->
<!-- Level 100 Pets -->
<div class="pieces">
    <div tabindex="0" data-pet-index="1" class="other-pet rich-item piece piece-mythic-bg piece-rarity-upgrade-1">
    <div class="piece-shine"></div>
    <div class="piece-hover-area" style="box-shadow: 0 0 6px #e041b7, 0 0 10px #e041b7, 0 0 12px #e041b7, 0 0 15px #e041b7, 0 0 18px #e041b7;"></div>
    <div style="background-image: url(&quot;/head/822d8e751c8f2fd4c8942c44bdb2f5ca4d8ae8e575ed3eb34c18a86e93b&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
    <div tabindex="0" data-pet-index="2" class="other-pet rich-item piece piece-legendary-bg piece-rarity-upgrade-0" style="box-shadow: 0 0 6px #f8b029, 0 0 10px #f8b029, 0 0 12px #f8b029, 0 0 15px #f8b029, 0 0 18px #f8b029;">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/64e22a46047d272e89a1cfa13e9734b7e12827e235c2012c1a95962874da0&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
    <div tabindex="0" data-pet-index="7" class="other-pet rich-item piece piece-epic-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area" style="box-shadow: 0 0 6px #d607d6, 0 0 10px #d607d6, 0 0 12px #d607d6, 0 0 15px #d607d6, 0 0 18px #d607d6;"></div>
    <div style="background-image: url(&quot;/head/5df4b3401a4d06ad66ac8b5c4d189618ae617f9c143071c8ac39a563cf4e4208&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
    <div tabindex="0" data-pet-index="16" class="other-pet rich-item piece piece-rare-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area" style="box-shadow: 0 0 6px #6f6ff6, 0 0 10px #6f6ff6, 0 0 12px #6f6ff6, 0 0 15px #6f6ff6, 0 0 18px #6f6ff6;"></div>
    <div style="background-image: url(&quot;/head/7e941987e825a24ea7baafab9819344b6c247c75c54a691987cd296bc163c263&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
    <div tabindex="0" data-pet-index="19" class="other-pet rich-item piece piece-uncommon-bg piece-rarity-upgrade-0" style="box-shadow: 0 0 6px #74cf74, 0 0 10px #74cf74, 0 0 12px #74cf74, 0 0 15px #74cf74, 0 0 18px #74cf74;">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/6eab75eaa5c9f2c43a0d23cfdce35f4df632e9815001850377385f7b2f039ce1&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
    <div tabindex="0" data-pet-index="20" class="other-pet rich-item piece piece-common-bg piece-rarity-upgrade-0" style="box-shadow: 0 0 6px #b3b3b3, 0 0 10px #b3b3b3, 0 0 12px #b3b3b3, 0 0 15px #b3b3b3, 0 0 18px #b3b3b3;">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/4e794274c1bb197ad306540286a7aa952974f5661bccf2b725424f6ed79c7884&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 100</div>
    </div>
</div>
<!-- Default Pets -->
<!-- Default Pets -->
<!-- Default Pets -->
<div class="pieces">
    <div tabindex="0" data-pet-index="1" class="other-pet rich-item piece piece-mythic-bg piece-rarity-upgrade-1">
    <div class="piece-shine"></div>
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/822d8e751c8f2fd4c8942c44bdb2f5ca4d8ae8e575ed3eb34c18a86e93b&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
    <div tabindex="0" data-pet-index="2" class="other-pet rich-item piece piece-legendary-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/64e22a46047d272e89a1cfa13e9734b7e12827e235c2012c1a95962874da0&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
    <div tabindex="0" data-pet-index="7" class="other-pet rich-item piece piece-epic-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/5df4b3401a4d06ad66ac8b5c4d189618ae617f9c143071c8ac39a563cf4e4208&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
    <div tabindex="0" data-pet-index="16" class="other-pet rich-item piece piece-rare-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/7e941987e825a24ea7baafab9819344b6c247c75c54a691987cd296bc163c263&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
    <div tabindex="0" data-pet-index="19" class="other-pet rich-item piece piece-uncommon-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/6eab75eaa5c9f2c43a0d23cfdce35f4df632e9815001850377385f7b2f039ce1&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
    <div tabindex="0" data-pet-index="20" class="other-pet rich-item piece piece-common-bg piece-rarity-upgrade-0">
    <div class="piece-hover-area"></div>
    <div style="background-image: url(&quot;/head/4e794274c1bb197ad306540286a7aa952974f5661bccf2b725424f6ed79c7884&quot;)" class="piece-icon item-icon custom-icon"></div>
    <div class="other-pet-level"><abbr title="Level">Lvl</abbr> 1</div>
    </div>
</div>
```

**Comment by nstringham** [comment]
> @nstringham i also did this, but idk how to implement it (lvl 100 pets background blur box-shadow)

That belongs in a completely separate PR my dude and if you don't know how to implement it It would probably be better suited as an issue

**Comment by nstringham** [comment]
> I cannot connect to MongoDB.

If you're having trouble connection to mongo I recommend using Atlas

File: public/resources/ts/themes.ts [hunk 3]
```diff
@@ -68,6 +68,9 @@ export function loadTheme(currentTheme: string): void {
           })
           .join(", ")})`;
         break;
+      case "linear-gradient":
+        value = 'linear-gradient(${background.angle}, ${background.color1} ${background.start}, ${background.color2} ${background.end})';
```
Comment (nstringham) [inline public/resources/ts/themes.ts:72]: literals should use ` not '
[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

**Review by nstringham — CHANGES_REQUESTED** [review]

**Comment by nstringham** [comment]
the gradient is very subtle it might be better to just use a solid gold color
image

**Review by JSanchoG — COMMENTED** [review]
done. tested

**Comment by JSanchoG** [comment]
> done. tested
<details>
<summary>fuck the gradient, all my homies hate gradient</summary>
gradient was pain
</details>

**Review by nstringham — CHANGES_REQUESTED** [review]
https://github.com/JSanchoG/SkyCrypt/pull/2

**Review by JSanchoG — COMMENTED** [review]
yup

**Comment by JSanchoG** [comment]
@nstringham im done with it, this is the final version.
and also with the second one https://github.com/SkyCryptWebsite/SkyCrypt/pull/596

**Comment by JSanchoG** [comment]
just wondering how long it would take to accept it

**Comment by nstringham** [comment]
This PR looks good from a technical perspective.

I don't understand why it's called blood Scorpion.

I don't personally like the way it looks

No maintainers have expressed interest in merging this theme

This Theme has zero thumbs up on [Discord](https://discord.com/channels/738971489411399761/752524295904821338/863647775101943808)

This PR has not been approved for CI workflow checks

**Comment by dukio** [comment]
Personally I like the idea of a red theme but I really don't like the background.... some time ago you posted a background with an enchantment table, I liked that one a lot more, do you still have it (to test)?

Also if you could upstream this PR to allow us to test it in localhost it would be great 👍 
(if you don't know how, just go to your [fork/branch](https://github.com/JSanchoG/SkyCrypt/tree/patch-2) and above to code click "Fetch upstream" -> "Fetch and merge")

**Comment by JSanchoG** [comment]
> Personally I like the idea of a red theme but I really don't like the background.... some time ago you posted a background with an enchantment table, I liked that one a lot more, do you still have it (to test)?
> 
> Also if you could upstream this PR to allow us to test it in localhost it would be great 👍
> (if you don't know how, just go to your [fork/branch](https://github.com/JSanchoG/SkyCrypt/tree/patch-2) and above to code click "Fetch upstream" -> "Fetch and merge")

done the fetch upstream thing,
old background with ench table: https://i.imgur.com/qI4oS4P.png

**Comment by JSanchoG** [comment]
> I don't understand why it's called blood Scorpion.

its not "blood scorpion" or like "scorpion of blood", its "bloody scorpion",
it symbolize the fall of the guild  i founded and grew myself called "ScorpionSquad" that quickly died after my false ban on hypixel (like the color of my mc skin and the right arm tattoo thing), it was just a big thing for me.

original name idea was created thanks to the hypixel skyblock update that implemented Scorpion Bow in it, i liked the way it sounds connected with "Squad" (bloody scorpion is the term that indicates guild's current state - its not dead, its also not in a good shape, its slowly dying but with the right attitude you can slowly work to revive it).

**Comment by nstringham** [comment]
> > I don't understand why it's called blood Scorpion.
> 
> its not "blood scorpion" or like "scorpion of blood", its "bloody scorpion",
> it symbolize the fall of the guild i founded and grew myself called "ScorpionSquad" that quickly died after my false ban on hypixel (like the color of my mc skin and the right arm tattoo thing), it was just a big thing for me.
> 
> original name idea was created thanks to the hypixel skyblock update that implemented Scorpion Bow in it, i liked the way it sounds connected with "Squad" (bloody scorpion is the term that indicates guild's current state - its not dead, its also not in a good shape, its slowly dying by bleeding (valuable players leaving) but with the right attitude and people you can slowly work to revive it).

I appreciate the simbolism but it seems like that will really only resonate with people who understand the refrance.

**Comment by JSanchoG** [comment]
> > > I don't understand why it's called blood Scorpion.
> > 
> > its not "blood scorpion" or like "scorpion of blood", its "bloody scorpion",
> > it symbolize the fall of the guild i founded and grew myself called "ScorpionSquad" that quickly died after my false ban on hypixel (like the color of my mc skin and the right arm tattoo thing), it was just a big thing for me.
> > 
> > original name idea was created thanks to the hypixel skyblock update that implemented Scorpion Bow in it, i liked the way it sounds connected with "Squad" (bloody scorpion is the term that indicates guild's current state - its not dead, its also not in a good shape, its slowly dying by bleeding (valuable players leaving) but with the right attitude and people you can slowly work to revive it).
> 
> I appreciate the simbolism but it seems like that will really only resonate with people who understand the refrance.

Bruh nobody cares why skycrypt is called skycrypt, hypixel is called hypixel, etc. people doesnt need to know origin of the name it just has to sound cool, do u think minecraft would be called minecraft if it didnt sound cool?

**Comment by JSanchoG** [comment]
What should i call it then? "Red theme"? Like why tf is candy cane theme called like that, nobody gives a shit im not gonna hate the theme for its name, but for authors creativity and original color combinations, which you limit so much, people simply are not able to express anything thru their theme

**Comment by WarpWing** [comment]
> > > I don't understand why it's called blood Scorpion.
> > 
> > 
> > its not "blood scorpion" or like "scorpion of blood", its "bloody scorpion",
> > it symbolize the fall of the guild i founded and grew myself called "ScorpionSquad" that quickly died after my false ban on hypixel (like the color of my mc skin and the right arm tattoo thing), it was just a big thing for me.
> > original name idea was created thanks to the hypixel skyblock update that implemented Scorpion Bow in it, i liked the way it sounds connected with "Squad" (bloody scorpion is the term that indicates guild's current state - its not dead, its also not in a good shape, its slowly dying by bleeding (valuable players leaving) but with the right attitude and people you can slowly work to revive it).
> 
> I appreciate the simbolism but it seems like that will really only resonate with people who understand the refrance.

No flame Nate but I think you're being a little bit critical. It doesn't matter where the name comes from but rather what it stands for. In essence, bloody scorpion is just a really cool name. So what if it's an inside joke, I think it's perfectly fine. It's a pretty cool name :D

**Comment by nstringham** [comment]
> > > > I don't understand why it's called blood Scorpion.
> > > 
> > > 
> > > its not "blood scorpion" or like "scorpion of blood", its "bloody scorpion",
> > > it symbolize the fall of the guild i founded and grew myself called "ScorpionSquad" that quickly died after my false ban on hypixel (like the color of my mc skin and the right arm tattoo thing), it was just a big thing for me.
> > > original name idea was created thanks to the hypixel skyblock update that implemented Scorpion Bow in it, i liked the way it sounds connected with "Squad" (bloody scorpion is the term that indicates guild's current state - its not dead, its also not in a good shape, its slowly dying by bleeding (valuable players leaving) but with the right attitude and people you can slowly work to revive it).
> > 
> > 
> > I appreciate the simbolism but it seems like that will really only resonate with people who understand the refrance.
> 
> No flame Nate but I think you're being a little bit critical. It doesn't matter where the name comes from but rather what it stands for. In essence, bloody scorpion is just a really cool name. So what if it's an inside joke, I think it's perfectly fine. It's a pretty cool name :D

yeah I get that It's just most of the themes have a name indicative of the theme itself

**Comment by nstringham** [comment]
Closing this PR because SkyCrypt doesn't really need more themes and the group of people interested in this Theme seems to be quite small. Additionally, this theme really looked better when it was using the enchantment table background which, unfortunately, doesn't fit in with other SkyCrypt themes.

In the long term future, I plan to add a system for third-party themes. see #597

In the short term future, Desktop Chrome users that want to use the Bloody Scorpion Theme should Download the [SkyCrypt Themer Chrome Extention](https://chrome.google.com/webstore/detail/skycrypt-themer/eeeichgfigdfdmnbmfbeggaolpnhbffc) and import Bloody_Scorpion_Theme.json ([MediaFire Download](https://www.mediafire.com/file/1bxevb82yqfm5ry/Bloody_Scorpion_Theme.json/file)) or copy the JSON below into a JSON file
```json
{"name":"Bloody Scorpion Theme","author":"Gozuk12","community":true,"images":{"bg":"https://i.imgur.com/qI4oS4P.png"},"backgrounds":{"skillbar":{"type":"color","color":"#ef325c"},"maxedbar":{"type":"color","color":"#f1a230"}},"colors":{"logo":"#ed1444","icon":"#ed1444","link":"#ef325c","hover":"#f15275","gold":"#f8b500"}}
```

**Pull request closed without merging**